UM/Qt/Bindings/ProfilesModel.py

```diff
@@ -93,31 +114,29 @@ def importProfile(self, url):
                 return { "status": "error", "message": catalog.i18nc("@info:status", "Failed to import profile from <filename>{0}</filename>: <message>{1}</message>", path, str(e)) }
             if profile: #Success!
                 profile.setReadOnly(False)
-                try:
-                    self._manager.addProfile(profile) #Add the new profile to the list of profiles.
-                except SettingsError.DuplicateProfileError as e:
-                    count = 2
-                    name = "{0} {1}".format(profile.getName(), count) #Try alternative profile names with a number appended to them.
-                    while self._manager.findProfile(name) != None:
-                        count += 1
-                        name = "{0} {1}".format(profile.getName(), count)
-                    profile.setName(name)
-                    self._manager.addProfile(profile)
-                    return { "status": "duplicate", "message": catalog.i18nc("@info:status", "Profile was imported as {0}", name) }
-                else:
-                    return { "status": "ok", "message": catalog.i18nc("@info:status", "Successfully imported profile {0}", profile.getName()) }
+                profile.setName(self._manager.makeUniqueProfileName(profile.getName())) #Ensure a unique name
+                if profile.getMachineTypeId():
+                    #Make sure the profile is available for the currently selected printer
+                    profile.setMachineTypeId(self._manager.getActiveMachineInstance().getMachineDefinition().getProfilesMachineId())
+                self._manager.addProfile(profile) #Add the new profile to the list of profiles.
+                return { "status": "ok", "message": catalog.i18nc("@info:status", "Successfully imported profile {0}", profile.getName()) }
 
         #If it hasn't returned by now, none of the plugins loaded the profile successfully.
         return { "status": "error", "message": catalog.i18nc("@info:status", "Profile {0} has an unknown file type.", path) }
 
-    @pyqtSlot(str, QUrl, str)
-    def exportProfile(self, name, url, fileType):
+    @pyqtSlot(int, str, QUrl, str)
+    def exportProfile(self, id, name, url, fileType):
         #Input checking.
         path = url.toLocalFile()
         if not path:
             return
 
-        profile = self._manager.findProfile(name)
+        if id==-1:
```

Comment:
Also, be sure to document cases like this well! Otherwise, the caller of the function will probably have no idea that -1 can be used to export the working profile.